source/known-hook-events.ts

```diff
@@ -1,8 +1,81 @@
-export default [
+import {IncomingMessage} from 'http';
+import {Options, CancelableRequest} from './utils/types';
+import {HTTPError} from './errors';
+
+export type InitHook = (options: Options) => void;
+// TODO: The `Error` type should confirm to any possible extended error type that can be thrown. See https://github.com/sindresorhus/got#hooksbeforeerror
+export type BeforeErrorHook = (error: Error) => Error | void | Promise<Error> | Promise<void>;
```

Comment:
I'm not sure why the `void` type is here... Shouldn't it be just `Error | Promise<Error>`?